compiler/rustc_expand/src/mbe/metavar_expr.rs

```diff
@@ -51,11 +51,18 @@ impl MetaVarExpr {
                 let mut result = Vec::new();
                 loop {
                     let is_var = try_eat_dollar(&mut iter);
-                    let element_ident = parse_ident(&mut iter, psess, outer_span)?;
+                    let token = parse_token(&mut iter, psess, outer_span)?;
                     let element = if is_var {
-                        MetaVarExprConcatElem::Var(element_ident)
+                        MetaVarExprConcatElem::Var(parse_ident_from_token(psess, token)?)
+                    } else if let token::TokenKind::Literal(token::Lit {
```

Comment:
```suggestion
                    } else if let token::Literal(token::Lit {
```
Could you avoid 3+ segment paths and use imports instead?
(For the stuff from token.rs all the enum variants already seem to be imported inside it.)

Reply:
Everything but `LitKind` (due to ambiguity) has been imported.